server/src/bots/telegram/services/messageRegistry.ts

```diff
@@ -56,4 +64,36 @@ class MessageRegistry {
 
 export const registry = new MessageRegistry();
 
-type CallBackQueryHandler = (bot: any, message: any, chatId: unknown) => any
+type CallBackQueryHandler = (bot: any, message: any, chatId: unknown, argument?: any) => any
+
+export function WithCommandArgument(handler: CallBackQueryHandler): any {
+    const context = registry;
+    return (bot: any, message: any, chatId: unknown, data?: string): any => {
+        try{
+            const argument = getArgfromMessage.call(context, data || message.text);
+            handler.call(context, bot, message, chatId, argument)
+        }
+        catch(Error){
+            logger.log('warn', {
+                ...message,
+                warningMessage: Error.message,
+            });
+            bot.sendMessage(chatId, Error.message);
+        }
+    }
+}
+function getArgfromMessage(messageText: string): string{
+    if(!messageText){
+        throw new Error('message could not be empty');
+    }
+    const execResult = this._singleParameterCommandRegex.exec(messageText);
+    if(!execResult){
+        return 'Plese input any command from avalable';
+    }
+    /* tslint:disable:no-string-literal */
```

Comment:
@jmelnich Thanks for adding lint rules.
I was just forced to disable one here since I believe it's kinda best approach to access regex group.
Let me know if we could handle it better. Since I know nothing about lint
🙇